nucypher/crypto/powers.py

```diff
@@ -99,7 +104,7 @@ def sign_message(self, message: bytes):
         signature = self.blockchain.interface.call_backend_sign(self.account, message)
         return signature
 
-    def verify_message(self, address: str, pubkey: bytes, message: bytes, signature: str):
+    def verify_message(self, address: str, pubkey: bytes, message: bytes, signature: Signature):
```

Comment:
This is a step in the wrong direction.

This is meant to be an interface to what the blockchain sees. Providing the `eth_keys` `Signature` object is only going to allow users to make mistakes with that.

This is meant to be an interface for exactly what a smart contract would most likely interact with. I'm not very keen on providing an object that we don't have control over.

Reply:
> Providing the eth_keys Signature object is only going to allow users to make mistakes with that.

...why?  Why doesn't it also allow them to do the right thing with that?

> I'm not very keen on providing an object that we don't have control over.

Wait, what?  Are you saying you are just wholesale against returning objects from dependency libraries?

I don't see how it makes *any* sense to return the hex.  We never use it that way.

Reply:
Essentially, I'm saying that `eth_keys` `Signature` class is too low-level to be exposed here. If we can restrict what can be done on this `Signature` class, then we could return that. Like we discussed in PDX, I would prefer this, but the ecosystem has yet to catch up here.

Reply:
I see. 

Well, I think that the proper pattern here is:

* Have address to verify
* Have message purporting to come from address
* Have signature against this message
* Validate signature
* Get public key from signature
* Derive address from signature
* Confirm that addresses match

That appears to be the intention of this object, and it performs this task very well.  So, do you want to cast to bytes and then cast back to `Signature`?  I guess that's fine.

Reply:
Yes, that would be ideal (for now). In the `verify_message` method, at least.